### PR TITLE
Revise package: pcsx-rearmed

### PR DESCRIPTION
**Pull request by Krush206**
00c802bcfb139c92615c4c716cdff97d0639cfc1: `-Oz` crashes the program;
6d1c9128e49cd0f75eed56ccb546ff589e425868: minimal optimizations and parameterizations.

### CODE REVIEW

File: x11-packages/pcsx-rearmed/build.sh [hunk 1]
```diff
@@ -23,20 +23,19 @@ termux_pkg_auto_update() {
 }
 
 termux_step_pre_configure() {
-	CFLAGS+=" $CPPFLAGS"
-	if [ "$TERMUX_ARCH" = "arm" ]; then
-		termux_setup_no_integrated_as
-	fi
+	CFLAGS+=' '"$CPPFLAGS"
+	CFLAGS="${CFLAGS/-Oz/-Os}"
+	[ "${TERMUX_ARCH}" = 'arm' ] && termux_setup_no_integrated_as
```
Comment (truboxl) [inline x11-packages/pcsx-rearmed/build.sh:28]: This `condition && command` isn't going to work when it's the last instruction in a bash function. Please use `if condition; then command; fi`. And please don't "over optimize" with single quotes.

Comment (TomJo2000) [reply x11-packages/pcsx-rearmed/build.sh:28]: Gonna have to disagree on the single quotes.
For 'static' strings they should be preferred.
But that's a personal pet peeve of mine.

The point about over optimization is much more important I feel like.
There is absolutely nothing wrong with the original version of this hunk.
Just add your `CFLAGS="${CFLAGS/-Oz/-Os}"` and be done with it.

File: x11-packages/pcsx-rearmed/build.sh [hunk 2]
```diff
@@ -23,20 +23,19 @@ termux_pkg_auto_update() {
...
-	install -Dm755 pcsx $TERMUX_PREFIX/bin/pcsx
-	mkdir -p $TERMUX_PREFIX/etc/pcsx $TERMUX_PREFIX/lib/pcsx_plugins
-	cp -r frontend/pandora/skin $TERMUX_PREFIX/etc/pcsx/
-	install -m755 plugins/*.so $TERMUX_PREFIX/lib/pcsx_plugins/
-	ln -fs ../../lib/pcsx_plugins $TERMUX_PREFIX/etc/pcsx/plugins
+	install -D -m755 pcsx ${TERMUX_PREFIX}/bin/pcsx
+	mkdir -p ${TERMUX_PREFIX}/etc/pcsx ${TERMUX_PREFIX}/lib/pcsx_plugins
+	cp -f -r frontend/pandora/skin ${TERMUX_PREFIX}/etc/pcsx/
+	install -m755 plugins/*.so ${TERMUX_PREFIX}/lib/pcsx_plugins/
+	ln -f -s ../../lib/pcsx_plugins ${TERMUX_PREFIX}/etc/pcsx/plugins
```
Comment (TomJo2000) [inline x11-packages/pcsx-rearmed/build.sh:40]: Love the braces personally.
But again, this doesn't fix anything.
You just unclumped the flags and added braces.

**Review by TomJo2000 — COMMENTED** [review]
My main problem is that this PR introduces 1 line of actual build script changes.
And a bunch of harmless but unnecessary noise which makes it unnecessarily hard to review.

I'm not qualified to speak on the patch changes, so I'll not mention them here.

**Comment by Krush206** [comment]
> I'm not qualified to speak on the patch changes, so I'll not mention them here.

Arrays are better optimized at compile-time than pointers are, @TomJo2000. Unlike pointers, arrays cannot be (re-)assigned; thus, no need of an additional `const` qualifier (i.e: `static const char *const arr = "abc"`). Since the path will never change, the `const` qualifier and `static` storage specifier make better compile-time optimizations. The `const` qualifier forbids (re-)assignment; the `static` storage specifier retains storage to the lifetime of the program.

However, do note aren't entirely optimizations. The switch of `@TERMUX_BASE_DIR@` to `@TERMUX_ANDROID_HOME@` is to set the program's initial search path to `$HOME`.

**Review by truboxl — APPROVED** [review]

**Pull request merged**